ivy/functional/frontends/numpy/linalg/matrix_and_vector_products.py

```diff
@@ -75,3 +75,10 @@ def multi_dot(arrays, *, out=None):
 def dot(a, b, out=None):
     a, b = promote_types_of_numpy_inputs(a, b)
     return ivy.matmul(a, b)
+
+@handle_numpy_out
+@to_ivy_arrays_and_back
+def einsum(
```

Comment:
In the frontend functions, it is essential to ensure that the function name and call signature precisely match the corresponding function in the target framework. The original function from the target framework is 
`def einsum(*operands, out=None, optimize=False, **kwargs)` [source](https://github.com/numpy/numpy/blob/v1.25.0/numpy/core/einsumfunc.py#L1010). Please refer to the frontend documentation [here](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html) for assistance. However, it appears that your call signature differs from the original function.